presto-main/src/main/java/com/facebook/presto/sql/planner/plan/JoinNode.java

```diff
@@ -194,4 +218,22 @@ public Symbol getRight()
             return right;
         }
     }
+
+    public boolean isDistributed()
+    {
+        return method == DISTRIBUTED;
+    }
+
+    public static boolean canDistributeJoin(Session session, PlanNode rightNode, List<JoinNode.EquiJoinClause> equiJoinClauses, JoinNode.Type type)
+    {
+        // The implementation of full outer join only works if the data is hash partitioned. See LookupJoinOperators#buildSideOuterJoinUnvisitedPositions
+        boolean isCrossJoin = type == Type.INNER && equiJoinClauses.isEmpty();
```

Comment:
Extract this to separate method.
Also what about `filter`. Doesn't it need to be `empty()`?